### PR TITLE
remove robot_model:: and robot_state::

### LINKED ISSUE #1878: prefer `moveit::core` over `robot_state` namespace
Confusingly, [robot_model/robot_model.h](https://github.com/ros-planning/moveit/blob/master/moveit_core/robot_model/include/moveit/robot_model/robot_model.h#L621) creates alias namespaces for `moveit::core`. Ioan added this long ago to stay backward compatible with old usage.

However, as a result the namespaces where not aligned throughout the codebase (grep for `robot_state::RobotState`, for example) and the alias is still there.

I believe we should deprecate the old names and align our codebases not to use the old names.

@davetcoleman

### PR DESCRIPTION
**Pull request by v4hn**
Fixes #1878 .

Please merge this as two commits, as we should only backport the internal usage, but not the API change.

### CODE REVIEW

**Comment by rhaschke** [comment]
Removing _robot_model_ and _robot_state_ requires changes in downstream packages like `moveit_visual_tools` as well.

**Comment by simonschmeisser** [comment]
Shouldn't there be a deprecation warning for some time before "removing" API? There are ***many*** places using `robot_state::RobotState` in ROS-I for example

**Comment by v4hn** [comment]
It's funny how @davetcoleman was the one complaining about the aliased namespaces, when his own projects use them. :smile: 

The problem with API deprecations in `master` is that it's not possible to bind the deprecation to a ROS release anymore, because OR wants us to abandon ROS.
In that context we agreed to allow more API-breaking changes in `master` directly.

Given the amount of usage of these namespaces, I switched to deprecation.
This will probably still fail with some ROS-I projects though, as I seem to recall they compile with `-Werror`(?).

**Comment by davetcoleman** [comment]
> It's funny how @davetcoleman was the one complaining about the aliased namespaces, when his own projects use them. smile

What projects specifically? I'm sure at one point I took some snippets from other places that used that type of namespace, yes. So what?

File: moveit_core/robot_model/include/moveit/robot_model/robot_model.h [hunk 1]
```diff
@@ -618,5 +618,5 @@ class RobotModel
 }  // namespace core
 }  // namespace moveit
 
-namespace robot_model = moveit::core;
-namespace robot_state = moveit::core;
+namespace[[deprecated("use moveit::core")]] robot_model = moveit::core;
```
Comment (rhaschke) [inline moveit_core/robot_model/include/moveit/robot_model/robot_model.h:621]: This syntax breaks Travis. Don't have the time to lookup the correct one, though.

**Comment by v4hn** [comment]
> What projects specifically

I noticed `moveit_visual_tools` with quite a few offenders.

But among other things the tutorials also use them a lot.

> This syntax breaks Travis.

True. It turns out namespace aliases (`namespace a = b`) do not support attributes at all, not even in the standard. Looks like an oversight to me, but might be related to simplified processing. Aliases are usually treated as straight text replacement.
A simple rewrite is `namespace [[deprecated]] a { using namespace b; }` which is allowed in the C++ standard.
**However**, because [the gcc developers are as slow to fix issues as we are](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=79817#c5), gcc did not support this syntax until **August 2019** (talking about c++11...). The patch did not even land in gcc 9.3, so it's still broken in Ubuntu 20.04.
Even the old-fashioned way of specifying gnu-specific deprecations (`__attribute__((deprecated))`) was not supported before.
Either way triggers
```
warning: 'deprecated' attribute directive ignored [-Wattributes]
```
and does not recognize the deprecation.

Clang just does the right thing.

That simply leaves us with *no way* of deprecating the namespaces without removing them.
Because the syntax is so common, let's just leave the namespace definitions around.
@henningkayser make sure you remove them for moveit2 though!

I added a decent hint to `MIGRATION.md` asking people to change the namespaces for future versions.

**Comment by henningkayser** [comment]
@v4hn can't we just deprecate the corresponding types using aliases (like https://github.com/ros-planning/moveit/pull/2127/commits/b7351456665a433a4d11866410e2617449cc41f5)

**Comment by v4hn** [comment]
Is the commit you cite actually accepted syntax by our CI? You didn't get CI to succeed yet.
After all it's also an alias declaration, even if for a class.

> @v4hn can't we just deprecate the corresponding types using aliases

We did similar things in the past for `MoveGroup` -> `MoveGroupInterface`, but in this case it would mean deprecating **every single class** in `moveit::core` for two namespaces.
That's fragile *and* a bulk of explicit definitions nobody will ever look at.
To me, that approach plus the inconvenience for users is not worth it to me to remove this hidden alias declaration...

**Comment by davetcoleman** [comment]
I see you already did the tutorials. Here's MVT: https://github.com/ros-planning/moveit_visual_tools/pull/65

**Pull request merged**